core/src/main/java/com/digitalpebble/stormcrawler/persistence/AdaptiveScheduler.java

```diff
@@ -237,7 +235,7 @@ public void init(Map stormConf) {
             // - old signature not copied
             // fall-back to DefaultScheduler
             LOG.debug("No signature for FETCHED page: {}", metadata);
-            if (setLastModified && signature != null && oldSignature == null) {
```

Comment:
this should not be removed

Reply:
`&& oldSignature == null` is redundant at this point and only costs an additional nullcheck because we can only get in this branch if either signature or oldSignature is null or both are null:

```java
if (Objects.equals(metadata.getFirstValue("fetch.statusCode"), "304")) {
...
} else if (signature == null || oldSignature == null) {
    // possible states: 
    // 1. signature IS NULL         AND oldSignature IS NOT NULL
    // 2. signature IS NOT NULL AND oldSignature IS NULL
    // 3. signature IS NULL         AND oldSignature IS NULL

    if(setLastModified && signature != null) {
        // signature is not null
        // Possible states left from above:
        // 2. signature IS NOT NULL AND oldSignature IS NULL
    }
}
```

Reply:
duhh, of course you are right. Need more coffee